features/net/FEATURE_IPV4/lwip-interface/lwip_stack.c

```diff
@@ -181,12 +186,19 @@ int lwip_bringup(void)
     return 0;
 }
 
-void lwip_bringdown(void)
+int lwip_bringdown(void)
 {
+    // Check if we've connected
+    if (!lwip_get_ip_address()) {
+        return NSAPI_ERROR_PARAMETER;
```

Comment:
It could be, there's not much precedent here that I am aware of.

I chose NSAPI_ERROR_PARAMETER to match the error on connecting an already connected network interface ([L148](https://github.com/ARMmbed/mbed-os/pull/2561/files#diff-266f114e2c179394a808cbde259fe946R148)). Disconnecting an already disconnected interface is more a user error than a network error.